Error while checking build/O2/fullCI for 410790c63c96440f9f0c32ab4252caefde74c19d at 2023-02-26 07:25:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/sm_20_atomic_functions.h:103:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/sm_20_atomic_functions.h:103:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: expected expression [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: type name does not allow function specifier to be specified [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: use of undeclared identifier 'noinline'; did you mean 'inline'? [clang-diagnostic-error]
/sw/SOURCES/O2/10835-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10835/410790c63c96440f9f0c32ab4252caefde74c19d/build_O2_fullCI/pretty.html).

--- a/Detectors/FIT/FDD/workflow/include/FDDWorkflow/RawDataReaderSpec.h
+++ b/Detectors/FIT/FDD/workflow/include/FDDWorkflow/RawDataReaderSpec.h
@@ -25,6 +25,7 @@
 #include "Framework/WorkflowSpec.h"
 #include "Framework/SerializationMethods.h"
 #include "DPLUtils/DPLRawParser.h"
+#include "DetectorsRaw/RDHUtils.h"
 
 #include <iostream>
 #include <vector>
@@ -52,9 +53,9 @@ class RawDataReaderSpec : public Task
     for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
       //Proccessing each page
       count++;
-      auto rdhPtr = it.get_if<o2::header::RAWDataHeader>();
+      auto rdhPtr = reinterpret_cast<const o2::header::RDHAny*>(it.raw());
       gsl::span<const uint8_t> payload(it.data(), it.size());
-      mRawReader.process(payload, rdhPtr->linkID, int(0));
+      mRawReader.process(payload, o2::raw::RDHUtils::getLinkID(rdhPtr), int(0));
     }
     LOG(info) << "Pages: " << count;
     mRawReader.accumulateDigits();

--- a/Detectors/FIT/FT0/workflow/CMakeLists.txt
+++ b/Detectors/FIT/FT0/workflow/CMakeLists.txt
@@ -35,6 +35,7 @@ o2_add_library(FT0Decoder
                        PUBLIC_LINK_LIBRARIES O2::DataFormatsFT0
                                      O2::Framework
                                      O2::DetectorsCommonDataFormats
+                                     O2::DetectorsRaw
                                      O2::DPLUtils
                                      TARGETVARNAME targetDecoderAVX512)
 target_compile_options(${targetDecoderAVX512} PRIVATE -mavx512f -mavx512vl -mavx512bw -mavx512dq)

--- a/Detectors/FIT/FT0/workflow/include/FT0Workflow/FT0DataReaderDPLSpec.h
+++ b/Detectors/FIT/FT0/workflow/include/FT0Workflow/FT0DataReaderDPLSpec.h
@@ -74,9 +74,9 @@ class FT0DataReaderDPLSpec : public Task
     for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
       //Proccessing each page
       count++;
-      auto rdhPtr = it.get_if<o2::header::RAWDataHeader>();
+      auto rdhPtr = reinterpret_cast<const o2::header::RDHAny*>(it.raw());
       gsl::span<const uint8_t> payload(it.data(), it.size());
-      mRawReader.process(payload, rdhPtr->linkID, rdhPtr->endPointID);
+      mRawReader.process(payload, o2::raw::RDHUtils::getLinkID(rdhPtr), o2::raw::RDHUtils::getEndPointID(rdhPtr));
     }
     LOG(info) << "Pages: " << count;
     mRawReader.accumulateDigits();

--- a/Detectors/FIT/FT0/workflow/src/FT0DataDecoderDPLSpec.cxx
+++ b/Detectors/FIT/FT0/workflow/src/FT0DataDecoderDPLSpec.cxx
@@ -12,6 +12,7 @@
 /// @file   FITDataDecoderDPLSpec.cxx
 
 #include "FT0Workflow/FT0DataDecoderDPLSpec.h"
+#include "DetectorsRaw/RDHUtils.h"
 #include <numeric>
 #include <emmintrin.h>
 #include <immintrin.h>
@@ -28,6 +29,13 @@ void FT0DataDecoderDPLSpec::run(ProcessingContext& pc)
 {
   auto t1 = std::chrono::high_resolution_clock::now();
 
+  auto dummyOutput = [&pc, this]() {
+    this->mVecDigits.resize(0);
+    pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginFT0, "DIGITSBC", 0, o2::framework::Lifetime::Timeframe}, mVecDigits);
+    this->mVecChannelData.resize(0);
+    pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginFT0, "DIGITSCH", 0, o2::framework::Lifetime::Timeframe}, mVecChannelData);
+  };
+
   // if we see requested data type input with 0xDEADBEEF subspec and 0 payload this means that the "delayed message"
   // mechanism created it in absence of real data from upstream. Processor should send empty output to not block the workflow
   {
@@ -43,10 +51,7 @@ void FT0DataDecoderDPLSpec::run(ProcessingContext& pc)
                dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, payloadSize,
                contDeadBeef == maxWarn ? fmt::format(". {} such inputs in row received, stopping reporting", contDeadBeef) : "");
         }
-        mVecDigits.resize(0);
-        pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginFT0, "DIGITSBC", 0, o2::framework::Lifetime::Timeframe}, mVecDigits);
-        mVecChannelData.resize(0);
-        pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginFT0, "DIGITSCH", 0, o2::framework::Lifetime::Timeframe}, mVecChannelData);
+        dummyOutput();
         return;
       }
     }
@@ -70,21 +75,34 @@ void FT0DataDecoderDPLSpec::run(ProcessingContext& pc)
     if (!it.size()) {
       continue; // excluding pages without payload
     }
-    auto rdhPtr = it.get_if<o2::header::RAWDataHeader>();
-    const uint16_t orbitTF = (rdhPtr->orbit) % 256;
+    auto* rdhPtr = reinterpret_cast<const o2::header::RDHAny*>(it.raw());
+    try {
+      int verRDH = o2::raw::RDHUtils::getVersion(rdhPtr);
+      if (verRDH < 5 || verRDH > o2::raw::RDHUtils::getVersion<o2::header::RDHHighest>()) {
+        LOGP(alarm, "Invalid RDH version {}, abandoning TF sending dummy output", verRDH);
+        dummyOutput();
+        return;
+      }
+    } catch (std::exception& e) {
+      LOG(alarm) << "Failed to extract RDH, abandoning TF sending dummy output, exception was: " << e.what();
+      dummyOutput();
+      return;
+    }
+    auto orb = o2::raw::RDHUtils::getHeartBeatOrbit(rdhPtr);
+    const uint16_t orbitTF = (orb) % 256;
     // const uint16_t feeID=rdhPtr->feeId;
-    arrOrbit[orbitTF] = rdhPtr->orbit;
-    const auto& linkID = rdhPtr->linkID;
-    const auto& endPoint = rdhPtr->endPointID;
+    arrOrbit[orbitTF] = orb;
+    const auto& linkID = o2::raw::RDHUtils::getLinkID(rdhPtr);
+    const auto& endPoint = o2::raw::RDHUtils::getEndPointID(rdhPtr);
     const uint16_t feeID = linkID + 12 * endPoint;
     if (feeID == mFEEID_TCM) {
       // Iterator is noncopyable, preparing RDH pointers and span objects
       arrOrbitSizePagesTCM[orbitTF] += it.size();
-      arrRdhTCMperOrbit[orbitTF].push_back(it.get_if<o2::header::RAWDataHeader>());
+      arrRdhTCMperOrbit[orbitTF].push_back(reinterpret_cast<const o2::header::RAWDataHeader*>(rdhPtr));
       arrDataTCMperOrbit[orbitTF].emplace_back(it.data(), it.size());
     } else {
       arrOrbitSizePages[orbitTF] += it.size();
-      arrRdhPtrPerOrbit[orbitTF][feeID].push_back(it.get_if<o2::header::RAWDataHeader>());
+      arrRdhPtrPerOrbit[orbitTF][feeID].push_back(reinterpret_cast<const o2::header::RAWDataHeader*>(rdhPtr));
       arrDataPerOrbit[orbitTF][feeID].emplace_back(it.data(), it.size());
     }
   }

